src/rust-crypto/backup.ts

```diff
@@ -164,9 +165,10 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         // We force a check to ensure to have the latest version. We also want to check that the backup is trusted
         // as we don't want to store the secret if the backup is not trusted, and eventually import megolm keys later from an untrusted backup.
         const backupCheck = await this.checkKeyBackupAndEnable(true);
+        const version = backupCheck?.backupInfo.version;
 
-        if (!backupCheck?.backupInfo?.version || !backupCheck.trustInfo.trusted) {
-            // There is no server-side key backup, or the backup is not signed by a trusted cross-signing key or trusted own device.
+        if (!backupCheck || !version) {
```

Comment:
There is no point in testing `backupCheck` as well as `version` here. `version` cannot be set unless `backupCheck` is set.

```suggestion
        if (!version) {
```